test/puppetlabs/trapperkeeper/services/status/status_service_test.clj

```diff
@@ -178,6 +188,28 @@
       (let [{:keys [status]} (http-client/get "http://localhost:8180/status/v1/services")]
         (is (= 200 status))))))
 
+(deftest status-check-error-handling-test
+  (with-status-service
+    app
+    [slow-service
+     broken-service]
+  (testing "handles case when a status check times out"
+    (with-redefs [puppetlabs.trapperkeeper.services.status.status-core/check-timeout (constantly 1)]
+      (let [resp (http-client/get "http://localhost:8180/status/v1/services/slow?level=critical")
+            body (json/parse-string (slurp (:body resp)))]
+        (is (= 503 (:status resp)))
+        (is (= "unknown"
+               (get body "state")))
+        (is (re-find #"timed out" (get body "status"))))))
+
+  (testing "handles case when a status check throws an exception"
+    (let [resp (http-client/get "http://localhost:8180/status/v1/services/broken?level=critical")
+            body (json/parse-string (slurp (:body resp)))]
+        (is (= 503 (:status resp)))
+        (is (= "unknown"
+               (get body "state")))
+        (is (re-find #"exception.*don't" (get body "status")))))))
+
```

Comment:
I think there should still be an explicit test of the behavior when a service returns a non-conformant status response, it just doesn't need a new service to do it.